### PR TITLE
Hunter/rate limit

### PR DESCRIPTION
**Pull request by hunter-bera**
Add rate limit lib with two types of stores: `memory` & `redis`

### CODE REVIEW

File: tools/limiter/limiter.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	RedisAddr        string
+	RedisClusterMode bool
+	ProxyCount       int
+}
+
+func New(config Config) *Limiter {
+	var lstore store.Store
+	switch config.Kind {
+	case "redis":
+		lstore = store.NewRedisStore(config.TTL, config.RedisAddr, config.RedisClusterMode)
```
Comment (calbera) [inline tools/limiter/limiter.go:32]: can you have it inferred so the `Kind` variable isn't necessary (enforces specific strings to be used). Just check if `RedisAddr != ""` then enable Redis, if not use mem

Comment (hunter-bera) [reply tools/limiter/limiter.go:32]: sure

**Comment by calbera** [comment]
can you also go into the `examples/listener/` app and implement a default Limiter with config and everyting enabled in app.go

**Comment by hunter-bera** [comment]
Added

**Review by calbera — DISMISSED** [review]

**Review by calbera — APPROVED** [review]

**Pull request merged**